app/internal/security/dependancies.py

```diff
@@ -0,0 +1,66 @@
+from typing import Union
```

Comment:
I believe some problems in this CR could be resolved with better naming.

We should also rethink a bit about the design: there are some functions that it's hard to tell why they're different one from another.